packages/openapi3/test/test-openapi-output.ts

```diff
@@ -378,6 +378,276 @@ describe("openapi3: definitions", () => {
   });
 });
 
+describe("openapi3: discriminated unions", () => {
+  it("defines unions with discriminators", async () => {
+    const openApi = await openApiFor(`                                                             
+      @discriminator("kind")                                                                       
+      model Pet { }                                                                                
+      model Cat extends Pet {                                                                      
+        kind: string;                                                                              
```

Comment:
this is exactly the kind of discriminator I don't understand.  How on the wire can I discriminate between types?  Seems like we should require kind: "Cat"  and kind: "Dog"

Reply:
As an _API style rule_ I agree.  But I don't believe that these should be encoded into the "generic" openapi3 emitter.

Reply:
I think I agree with the confusion. Maybe an example of how a generic string being a discriminator gets used in practice would help? What do existing openapi code generators give you on e.g. typescript for this pattern?

Reply:
A generic string is the "catch-all".  It lets me add lizard later if I want.

But I suppose restricting this now and then possibly loosening it up later is better than starting loose and then being unable to tighten later.  So I will make the change to require string constants and we'll see if we get feedback asking for generic string.

Reply:
An interesting twist here -- without subtype support, this preclude more than one level of inheritance.  E.g.:
```
      model Pet {
        name: string;
        weight?: float32;
      }                                                                                
      model Cat extends Pet {                                                                      
        kind: "cat";                                                                              
        meow: int32;                                                                               
      }                                                                                            
      model Dog extends Pet {                                                                      
        kind: "dog";                                                                              
        bark: string;                                                                              
      }
      model Beagle extends Dog {
        kind: "beagle";
        purebred: boolean;
      } 
```

I suppose that for now we can restrict this too, but I'm pretty sure folks will want multi-level inheritance.

Reply:
I don't yet understand the connection between this and multiple levels. It feels like https://github.com/microsoft/cadl/pull/89#issuecomment-983775654 shows a pattern for multiple levels where the set of types is still closed?